### PR TITLE
Serve correct grpc listener

### PR DESCRIPTION
**Pull request by lampajr**
## Description

This change includes the folowing fixes:
* Set correct `grpcListener` on grpcServer serve invocation
* Remove useless listener parameter on `grpcListenerServer` and `graphQlListenerServer`

## How Has This Been Tested?

This has been tested starting the grpc server using `make run/server` and executing `test/python/test_mlmetadata.py`, without this change the script was stuck at the first grpc call, whereas with this change the script completes.

## Merge criteria:

- [x] The commits and have meaningful messages; the author will squash them [after approval](https://github.com/opendatahub-io/opendatahub-community/blob/main/contributor-cheatsheet.md#:~:text=Usually%20this%20is%20done%20in%20last%20phase%20of%20a%20PR%20revision) or will ask to merge with squash.
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: cmd/serve.go [hunk 1]
```diff
@@ -102,15 +103,15 @@ func runServer(cmd *cobra.Command, args []string) error {
...
 	// GraphQL server
-	gqlServer := graphQlListenerServer(grpcListener, db)
+	gqlServer := graphQlListenerServer(db)
 
 	// start cmux listeners
 	g := new(errgroup.Group)
 	g.Go(func() error {
 		glog.Info("starting gRPC server...")
-		return grpcServer.Serve(listener)
+		return grpcServer.Serve(grpcListener)
```
Comment (tarilabs) [inline cmd/serve.go:114]: 🔭

**Review by tarilabs — APPROVED** [review]
seen live, agree; well spotted (imho) ! 👍

**Review by antisaling — APPROVED** [review]
lgtm!

**Review by dhirajsb — CHANGES_REQUESTED** [review]
The listener is not useless. :smile:  It's meant to filter gRPC calls.
The server is being _cute_ by using the same port for both gRPC and GraphQL calls. 

I've opened another issue #22  to change the mux to gin and use two separate ports for gRPC and GraphQL to avoid the filtering issue altogether.

**Comment by lampajr** [comment]
> The listener is not useless. :smile:  It's meant to filter gRPC calls.
> The server is being _cute_ by using the same port for both gRPC and GraphQL calls. 
> 
> I've opened another issue #22  to change the mux to gin and use two separate ports for gRPC and GraphQL to avoid the filtering issue altogether. 

I meant it was useless in those functions, the parameter (listener) was not used at all - that's why I removed them.

If you agree I would merge this PR as this fixes the actual behavior (at least we stabilize the server), then I will work on the issue you created as separated work. wdyt?

**Review by dhirajsb — APPROVED** [review]
lgtm for now
If you can take on the new issue to switch to gin-gonic that'll solve any potential issues with grpc and cmux.

**Comment by lampajr** [comment]
> lgtm for now
> If you can take on the new issue to switch to gin-gonic that'll solve any potential issues with grpc and cmux. 

thanks @dhirajsb ! 
Sure i will take that issue as well 💪

**Pull request merged**